runtime/src/prioritization_fee_cache.rs

```diff
@@ -438,9 +473,8 @@ mod tests {
 
         // wait till update is done
         while block_fee
-            .lock()
-            .unwrap()
-            .get_min_transaction_fee()
+            .get(&bank.bank_id())
+            .map(|block_fee| block_fee.get_min_transaction_fee())
```

Comment:
calling `block_fee.get_min_transaction_fee()` isn't meaningless... i think you meant to call `.and_then()`? or remove `.map(...)` altogether?

Reply:
i mean, i think `.map(...)`'s return type is `Option<Option<u64>>`. so, we aren't checking `.is_none()` on the returned value of `.get_min_transaction_fee()`.

Reply:
Good catch! Yea, should use `and_then`; then realized just checking if `block_fee.get_min_transaction_fee()` has value would make test flaky. In case a `bank` updates cache with multiple transactions, since each transaction applies to cache one at time, the `sync_update()` should make sure all transactions have applied (instead of checking if _any_ transaction has applied). I moved `metrics.accumulate_successful_transaction_update_count(1);` and _conveniently_ used it to sync updates. tbh, not too happy double-purposing metrics counter here.

Reply:
oh, glad that we fixed test flakiness. :)

> tbh, not too happy double-purposing metrics counter here.

same here. but i think this is acceptable, considering this is just a tiny piece of test harnesses...